core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/AddExchangesBelowPartialAggregationOverGroupIdRuleSet.java

```diff
@@ -104,25 +104,33 @@
             typeOf(ExchangeNode.class)
                     .with(scope().equalTo(REMOTE))
                     .with(source().matching(
-                            // PushPartialAggregationThroughExchange adds a projection. However, it can be removed if RemoveRedundantIdentityProjections is run in the mean-time.
-                            typeOf(ProjectNode.class).capturedAs(PROJECTION)
+                            typeOf(ProjectNode.class)
```

Comment:
any ProjectNode is fine?

Reply:
My understanding is that one of the rules adds identity projections that are later removed. This rule is trying to capture these projections. @findepi could you please take a look and validate whether my understanding is correct?